examples/mediadevices/src/index.js

```diff
@@ -119,20 +119,22 @@ function participantDisconnected(participant) {
   participantDiv.parentNode.removeChild(participantDiv);
 }
 
-// reads selected audio input, and updates preview and room to use the device.
+// // reads selected audio input, and updates preview and room to use the device.
 async function applyAudioInputDeviceChange(event) {
-  var audio = document.querySelector('audio#audioinputpreview');
-  var waveformContainer = document.querySelector('div#audioinputwaveform');
+  const waveformContainer = document.querySelector('div#audioinputwaveform');
+  let localAudioTrackPublication = null;
   if (event) {
     event.preventDefault();
     event.stopPropagation();
   }
-
-  await applyAudioInputDeviceSelection(deviceSelections.audioinput.value, audio, someRoom);
-
-  if (audio.srcObject) {
-    var canvas = waveformContainer.querySelector('canvas');
-    waveform.setStream(audio.srcObject);
+  if(someRoom) {
+    [localAudioTrackPublication] = Array.from(someRoom.localParticipant.audioTracks.values());
+  }
+  if (localAudioTrackPublication) {
```

Comment:
This still doesn't work if you want to change and apply devices before joining a Room.